pytorch_lightning/callbacks/progress.py

```diff
@@ -0,0 +1,371 @@
+    - **sanity check progress:** the progress during the sanity check run
+    - **main progress:** shows training + validation progress combined. It also accounts for
+      multiple validation runs during training when
+      :paramref:`~pytorch_lightning.trainer.trainer.Trainer.val_check_interval` is used.
+    - **validation progress:** only visible during validation;
+      shows total progress over all validation datasets.
+    - **test progress:** only active when testing; shows total progress over all test datasets.
+
+    For infinite datasets, the progress bar never ends.
+
+    If you want to customize the default ``tqdm`` progress bars used by Lightning, you can override
+    specific methods of the callback class and pass your custom implementation to the
+    :class:`~pytorch_lightning.trainer.trainer.Trainer`:
+
+    Example::
+
+        class LitProgressBar(ProgressBar):
+
+            def init_validation_tqdm(self):
+                bar = super().init_validation_tqdm()
+                bar.set_description('running validation ...')
+                return bar
+
+        bar = LitProgressBar()
+        trainer = Trainer(callbacks=[bar])
+
+    Args:
+        refresh_rate:
+            Determines at which rate (in number of batches) the progress bars get updated.
+            Set it to ``0`` to disable the display. By default, the
+            :class:`~pytorch_lightning.trainer.trainer.Trainer` uses this implementation of the progress
+            bar and sets the refresh rate to the value provided to the
+            :paramref:`~pytorch_lightning.trainer.trainer.Trainer.progress_bar_refresh_rate` argument in the
+            :class:`~pytorch_lightning.trainer.trainer.Trainer`.
+        process_position:
+            Set this to a value greater than ``0`` to offset the progress bars by this many lines.
+            This is useful when you have progress bars defined elsewhere and want to show all of them
+            together. This corresponds to
+            :paramref:`~pytorch_lightning.trainer.trainer.Trainer.process_position` in the
+            :class:`~pytorch_lightning.trainer.trainer.Trainer`.
+
+    """
+    def __init__(self, refresh_rate: int = 1, process_position: int = 0):
+        super().__init__()
+        self._refresh_rate = refresh_rate
+        self._process_position = process_position
+        self._enabled = True
+        self.main_progress_bar = None
+        self.val_progress_bar = None
+        self.test_progress_bar = None
+
+    def __getstate__(self):
+        # can't pickle the tqdm objects
+        state = self.__dict__.copy()
+        state['main_progress_bar'] = None
+        state['val_progress_bar'] = None
+        state['test_progress_bar'] = None
+        return state
+
+    @property
+    def refresh_rate(self) -> int:
+        return self._refresh_rate
+
+    @property
+    def process_position(self) -> int:
+        return self._process_position
+
+    @property
+    def enabled(self) -> bool:
```

Comment:
```suggestion
    def is_enabled(self) -> bool:
```
as it does not change the state and not easy to overlook with `enable` without "d"